### PR TITLE
UI: Fullscreen toggle does not work when all panels are collapsed

### PR DESCRIPTION
**Pull request by Tomastomaslol**
Issue: #11782 

## What I did
* When exiting `isFullScreen` and navigation is not show set `showNav` to `true`
* When `togglePanel` is invoked and the panel should be hidden and navigation is hidden. Set `isFullScreen` to `true`
*  When `toggleNav ` is invoked and the navigation should be hidden and the panel is hidden. Set `isFullScreen` to `true`

## How to test

The expected behaviour is described here: https://github.com/storybookjs/storybook/issues/11782#issuecomment-674435108

### Scenario 1
* Hide `Panel` and `Nav`
* _Should Enter fullscreen mode automatically_
* Exit fullscreen mode
* Navigation should be shown 

Kapture 2020-08-17 at 9 42 09

### Scenario 2
* Hide `Nav`
* Enter fullscreen mode
* Exit fullscreen mode
* Navigation and panel should be shown

Kapture 2020-08-17 at 9 50 42

### Scenario 3
* Hide `Panel`
* Enter fullscreen mode
* Exit fullscreen mode
* Navigation should be shown

### Scenario 4
* Enter fullscreen mode
* Exit fullscreen mode
* Navigation and panel should be shown

### CODE REVIEW

**Comment by yannbf** [comment]
Awesome work @Tomastomaslol !!!

I'm just wondering about scenario 3:

> Hide Panel
> Enter fullscreen mode
> Exit fullscreen mode
> Navigation should be shown


If we check @domyen's explanation:

> When you press the close button in full screen mode, it should open the sidebar. **If the addon panel was previous open it should take you back to it's previous state.** If the previous state of the sidebar or addon panel can't be determined, reset the manager UI.

So in this case, at least from what I read, when exiting full screen mode, having the panel being hidden previously, it should go back to the previous state, which is panel open, navigation hidden. Please correct me if I'm wrong @domyen.

Also, there was also a nice suggestion to give a hint of the shortcuts in the buttons:
image
"Go full screen [F]"

image
"Change addon orientation [D]
"Hide addons [A]"

Would you be so kind to add that as well?

File: lib/ui/src/components/panel/panel.tsx [hunk 1]
```diff
@@ -59,14 +59,14 @@ const AddonPanel = React.memo<{
...
-          title="Change orientation"
+          title="Change addon orientation [D]"
         >
           <Icons icon={panelPosition === 'bottom' ? 'bottombar' : 'sidebaralt'} />
         </DesktopOnlyIconButton>
         <DesktopOnlyIconButton
           key="visibility"
           onClick={actions.toggleVisibility}
-          title="Hide addons"
+          title="Hide addons [A]"
```
Comment (ndelangen) [inline lib/ui/src/components/panel/panel.tsx:69]: these are not hard coded, but rather user configurable!

**Review by ndelangen — CHANGES_REQUESTED** [review]
hotkeys are configurable

File: lib/ui/src/components/panel/panel.tsx (hunk 1 above)
Comment (yannbf) [reply lib/ui/src/components/panel/panel.tsx:69]: I assume to get them it's kinda like this?

```
import { Consumer } from '@storybook/api';

const Component = () => (
  <Consumer>
    {({ api }) => {
      const shortcuts = api.getShortcutKeys();
      return (
        <>
          <ComponentForFullScreen title={`Toggle fullscreen [${shortcuts.fullScreen}]`} />;
          <ComponentForAddonOrientation
            title={`Change addon orientation [${shortcuts.panelPosition}]`}
          />
          ;
          <ComponentForHidingAddons title={`Hide addons [${shortcuts.togglePanel}]`} />;
        </>
      );
    }}
  </Consumer>
);
```

Or is there another way to get them?

Comment (Tomastomaslol) [reply lib/ui/src/components/panel/panel.tsx]: Thanks for the feedback! I fixed this issue in my latest commit please let me what you think of my proposed solution.

**Comment by yannbf** [comment]
It all seems good to me!
@domyen could you please double check this PR, especially [this comment](https://github.com/storybookjs/storybook/pull/11810#issuecomment-678588883)? Thanks!

**Comment by shilman** [comment]
@Tomastomaslol it looks like CI is failing. can you take a look?

```
Error: Evaluation failed: TypeError: Cannot read property 'getShortcutKeys' of undefined
    at Object.PanelMapper [as current] (https://5a375b97f4b14f0020b0cda3-chgwvaljgf.chromatic.com/main.2dee079d08dfc61f7a62.bundle.js:1:462099)
    at ManagerConsumer (webpack://storybook_docs_dll//tmp/storybook/lib/api/dist/index.js?:624:23)
```

**Comment by domyen** [comment]
Will have a chance to take a look today. Thanks for the heads up Yann.

**Comment by domyen** [comment]
@Tomastomaslol thanks so much for this PR! 🙇‍♂️

The gifs are super helpful.

> If the addon panel was previous open it should take you back to it's previous state.

> Does this mean the state previous to when the full-screen mode was toggled? If so I believe the my suggested implementation is correct.

✅ I think you got it. The deploy previews are down so it's tough to verify the behavior. 

✅Example A:
1. Rest state is the sidebar and addon panel are shown
2. User goes full screen, sidebar and addon panel are hidden
3. User exits full screen, sidebar and addon panel are shown

✅Example B:
1. Rest state is the sidebar is shown and _addon panel is hidden_
2. User goes full screen, sidebar is hidden and _addon panel continues to be hidden_
3. User exits full screen, sidebar is shown and _addon panel continues to be hidden_

Also, thank you @yannbf for guiding this issue and PR home 🙏

**Comment by Tomastomaslol** [comment]
@shilman Thanks for pointing out the failing tests. In the future, I will pay more attention to checks and its outcomes to make sure it hasn't flagged any issues in my PR.

While fixing the tests I realised there were better ways to structure my code so I did some refactoring. There should be no change to functionality. 

@domyen I appreciate you taking time look at my code and helping me verify the behaviour!

And thanks for all help @yannbf. I had a lot of fun working on this! 🙂

**Comment by domyen** [comment]
This is ready to go from my end pending CI passing 👍

**Pull request merged**